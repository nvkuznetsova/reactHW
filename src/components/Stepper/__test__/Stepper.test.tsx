import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Stepper, StepperProps } from '..';

describe('Stepper', () => {
  const increment = jest.fn();
  const decrement = jest.fn();
  const getProps = (): StepperProps => ({
    value: 1,
    increment,
    decrement,
  });
  const setup = (props = {}): ShallowWrapper => (
    shallow(<Stepper {...getProps()} {...props}/>)
  );

  afterEach(jest.clearAllMocks);
  
  it('should match snapshot', () => {
    const component = setup();
    expect(component).toMatchSnapshot()
  });

  it('should call increment on click', () => {
    const component = setup();
    const incrementBtn = component.find('[data-marker="increment"]');
    incrementBtn.simulate('click');
    expect(increment).toBeCalled();
  });

  it('should call decrement on click', () => {
    const component = setup();
    const decrementBtn = component.find('[data-marker="decrement"]');
    decrementBtn.simulate('click');
    expect(decrement).toBeCalled();
  });
});