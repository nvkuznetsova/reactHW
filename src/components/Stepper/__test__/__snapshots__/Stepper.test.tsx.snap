// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper should match snapshot 1`] = `
<div
  className="stepper border-primary text-primary"
>
  <span
    data-marker="decrement"
    onClick={[MockFunction]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f068",
            "M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
          ],
          "iconName": "minus",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
  <span>
    1
  </span>
  <span
    data-marker="increment"
    onClick={[MockFunction]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f067",
            "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
          ],
          "iconName": "plus",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
</div>
`;
